source/extensions/filters/http/oauth2/oauth_client.cc

```diff
@@ -65,14 +68,14 @@ void OAuth2ClientImpl::asyncGetAccessToken(const std::string& auth_code,
   request->headers().setContentLength(body.length());
   ENVOY_LOG(debug, "Dispatching OAuth request for access token.");
   dispatchRequest(std::move(request));
-
-  ASSERT(state_ == OAuthState::Idle);
-  state_ = OAuthState::PendingAccessToken;
 }
 
 void OAuth2ClientImpl::asyncRefreshAccessToken(const std::string& refresh_token,
                                                const std::string& client_id,
                                                const std::string& secret, AuthType auth_type) {
+  ASSERT(state_ == OAuthState::Idle);
```

Comment:
Did you  add, or is there an existing test for immediate response in case of refresh token request?

Reply:
Added one now